cypress/integration/control_dynamiclink.js

```diff
@@ -28,7 +28,7 @@ context('Dynamic Link', () => {
 		cy.new_form('Test Dynamic Link');
 		cy.get('form > [data-fieldname="doc_type"]').type('User');
 		cy.get('form > [data-fieldname="doc_id"]').click();
-		cy.get('[id="awesomplete_list_4"]').its('length').should('be.gte', 0);
+		cy.get_field("doc_id").its('length').should('be.gte', 0);
```

Comment:
This is not correct. Here we are just getting the field... We should check if the dropdown elements are populated or not...

My [previously suggested change](https://github.com/frappe/frappe/pull/16350#discussion_r830933229) did not work?

Reply:
Nope...that didnt work....its not taking the role "listbox"

Reply:
You can try
```suggestion
		cy.get_field("doc_id").find("li").should('be.gte', 0);
```

Reply:
The one which you have suggested wont work bcoz doc_id field does not have "li" element connected to it...
Tried writing it in the below way. Check it out once if the below is fine.
cy.get('.awesomplete').find("li").its('length').should('be.gte', 0);

Reply:
Still, the concern your suggestion is that cypress will select any ".awesomplete" component it finds on the page... it is not specific yet (which can cause flaky behaviour/ improper testing)

let's make it more specific.
```
cy.get('[data-fieldname="doc_id"]').find('.awesomplete').find("li").its('length').should('be.gte', 0);
```

This will ensure that only `.awesomplete` component of `doc_id` field is selected.